### PR TITLE
Policy refactor

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
> tACK, i can't see a problem. However I could not understand why this change was needed. Can you summarize shortly?

Thanks. Have you checked the commit messages? The script configs preprocess commit has a long explanation. The others are to prepare adding a `Tr(...)` Taproot enum entry next to `Wsh()` in `ParsedPolicy`, as Taproot policies will need re-use this.

**Comment by asi345** [comment]
Oh you're right. Sorry I just missed the commit message paragraphs. Thanks!
